flow/probes/sflow.go

```diff
@@ -190,3 +183,136 @@ func NewSFlowProbe(a string, p int, g *graph.Graph) (*SFlowProbe, error) {
 
 	return probe, nil
 }
+
+type SkydivePacket struct {
+	gopacket.Packet
+	ci     gopacket.CaptureInfo
+	raw    *sflow.RawPacketFlow
+	sample *sflow.FlowSample
+}
+
+type SFlowReader struct {
+	gopacket.PacketSource
+	ProbeTarget      string
+	PacketChannel    chan SkydivePacket
+	RawPacketCounter uint64
+}
+
+func (r *SFlowReader) LinkType() layers.LinkType {
+	return layers.LinkTypeEthernet
+}
+
+func (r *SFlowReader) CreatePacket(sample *sflow.FlowSample, raw *sflow.RawPacketFlow) {
+	var ci gopacket.CaptureInfo
+	ci.Timestamp = time.Now()
+	ci.CaptureLength = int(raw.FrameLength)
+	ci.Length = int(raw.HeaderSize)
+
+	data := raw.Header
+	r.PacketChannel <- SkydivePacket{
+		gopacket.NewPacket(data, r.LinkType(), gopacket.DecodeOptions{}),
+		ci,
+		raw,
+		sample}
+}
+
+/* PacketSource */
+func (r *SFlowReader) NextPacket() (SkydivePacket, error) {
+	packet := <-r.PacketChannel
+	m := packet.Metadata()
+	m.CaptureInfo = packet.ci
+	m.Truncated = m.Truncated || packet.ci.CaptureLength < packet.ci.Length
+
+	return packet, nil
+}
+
+func (p *SFlowReader) packetsToChannel() {
+	defer close(p.PacketChannel)
+	for {
+		packet, err := p.NextPacket()
+		if err == io.EOF {
+			return
+		} else if err == nil {
+			p.PacketChannel <- packet
+		}
+	}
+}
+
+func (p *SFlowReader) Packets() chan SkydivePacket {
+	if p.PacketChannel == nil {
+		p.PacketChannel = make(chan SkydivePacket, 1000)
+		go p.packetsToChannel()
```

Comment:
I do not understand the purpose of this go-routine since we are already in a go-routine started by the Start function. If I understand correctly we have a publisher of packets which is the Parse function and a consumer function which is asyncStart ? so from my point of view having only one go-routine the "Start" one should be enough ?